src/gui/GuiApplication.cpp

```diff
@@ -104,6 +105,16 @@ GuiApplication::GuiApplication()
 	QApplication::setAttribute(Qt::AA_DontShowIconsInMenus, true);
 #endif
 
+#ifdef LMMS_HAVE_SUIL
+#ifdef LMMS_BUILD_APPLE
+	// Ensure running from bundle
+	if(qApp->applicationDirPath().endsWith("/Contents/MacOS")) {
+		// Inform Suil to load modules from a bundled application
+		qputenv("SUIL_MODULE_DIR", qApp->applicationDirPath().append("/../Frameworks/suil-0/").toUtf8());
+	}
+#endif
+#endif // LMMS_HAVE_SUIL
+
```

Comment:
> > Initial testing on MacOS is OK but we don't use a launcher script for MacOS, Perhaps [qputenv()](https://stackoverflow.com/a/45348742/3196753) combined with qApp->applicationDirPath() + "../Frameworks/suil-0/? Thoughts welcome
> 
> Done via [5bb73b0](https://github.com/LMMS/lmms/commit/5bb73b0c9b7f92f4d6490ad51ea38cc45fe1b195). Feel free to move the code elsewhere if needed.

I asked this question on Discord too, but should we embrace this type of logic for Linux too?

* `*squashfs_root*`
* `*.mount_lmms*`
* `/opt/lmms/` (futureproof for when #7252 is merged)

Reply:
Quoting @JohannesLorenz on discord:

> I would prefer that. Especially, since you can then test if it has been set by the user, and in this case not override it.